lib/jekyll/commands/serve.rb

```diff
@@ -94,7 +90,7 @@ def start(opts)
           if Jekyll.env == "development"
             config["url"] = default_url(config)
           end
-          [Build, Serve].each { |klass| klass.process(config) }
+          process_with_graceful_fail("serve", config, Build, Serve)
```

Comment:
The first argument expected here is not a `String` but the `cmd` object from `prog.command(:serve) do |cmd|`
This is the reason I added a `friction` label to this PR..

Reply:
I see. So, we need to pass `cmd` from `init_with_program()` to `start()`?

Reply:
Either that or do away with `start` entirely.. i.e. merge `start` into `init_with..`

Reply:
I imagine it was added to reduce perceived complexity with regard to Rubocop. `start()` should be a private method, and I don't see any problem passing `cmd` to it.

Reply:
Yeah, we have to decide between _changing method signature_ vs _doing away with unnecessary method_

Reply:
Oh, I'm not worried about backwards compatibility on this one. I can't imagine anybody else making use of `start()`

If you think it makes more sense to inline it, and Rubocop doesn't complain, I'm :+1::+1: